app/src/main/java/com/io/movio/domain/GetMoviesUseCase.kt

```diff
@@ -1,9 +1,10 @@
 package com.io.movio.domain
 
-import com.io.movio.data.repositories.MovieRepository
+import com.io.movio.data.repositories.MovieRepositoryImpl
+import javax.inject.Inject
 import kotlin.Exception
 
-class GetMoviesUseCase(private val movieRepository: MovieRepository = MovieRepository) : UseCase< Unit , Result<List<Movie>>> {
+class GetMoviesUseCase @Inject constructor(private val movieRepository: MovieRepositoryImpl) : UseCase< Unit , Result<List<Movie>>> {
```

Comment:
This should be `private val movieRepository: MovieRepository` and Hilt will provide you the actual instance - MovieRepositoryImpl. That's the whole idea of DI. You expect an object following a contract. You just know of the contract and don't know what object and from where it comes from